File: rehovot-ads-packages/ios/Plugin/BannerManager.swift
import Foundation

// Banner responsibilities are isolated so future SDK adapters can replace them cleanly.
final class BannerManager {
    private let logger: Logger
    private var isLoaded: Bool = false

    init(logger: Logger) {
        self.logger = logger
    }

    func show(_ options: BannerRequest) {
        logger.debug("TODO: Show Banner")
        logger.debug("Banner options: \(options)")
        isLoaded = false
    }

    func hide() {
        logger.debug("TODO: Hide Banner")
    }

    func destroy() {
        logger.debug("TODO: Destroy Banner")
        isLoaded = false
    }

    func loaded() -> Bool {
        return isLoaded
    }
}

File: rehovot-ads-packages/android/src/main/java/com/rehovottech/levelplay/BannerManager.kt
package com.rehovottech.levelplay

// Banner concerns stay isolated so future SDK adapters can replace only this file.
class BannerManager(
    private val logger: Logger,
) {
    private var isLoaded: Boolean = false

    public fun show(options: BannerRequest) {
        logger.d("TODO: Show Banner")
        logger.d("Banner options: $options")
        isLoaded = false
    }

    public fun hide() {
        logger.d("TODO: Hide Banner")
    }

    public fun destroy() {
        logger.d("TODO: Destroy Banner")
        isLoaded = false
    }

    public fun isLoaded(): Boolean {
        return isLoaded
    }
}
